python/tvm/runtime/graph_runtime_factory.py

```diff
@@ -0,0 +1,144 @@
+    """
+    if not isinstance(graph_json_str, string_types):
+        try:
+            graph_json_str = graph_json_str._tvm_graph_json()
+        except AttributeError:
+            raise ValueError("Type %s is not supported" % type(graph_json_str))
+    fcreate = get_global_func("tvm.graph_runtime_factory.create")
+    args = []
+    for k, v in params.items():
+        args.append(k)
+        args.append(ndarray.array(v))
+    return GraphRuntimeFactoryModule(fcreate(graph_runtime_kind, graph_json_str, libmod, module_name, *args))
+
+
+class GraphRuntimeFactoryModule(Module):
+    """Graph runtime factory module.
+
+    This is a module of graph runtime factory
+
+    Parameters
+    ----------
+    module : Module
+        The interal tvm module that holds the actual graph functions.
+
+    Attributes
+    ----------
+    module : Module
+        The interal tvm module that holds the actual graph functions.
+    """
+
+    def __init__(self, module):
+        self.module = module
+        self._select_module = module["select_module"]
+        self._import_module = module["import_module"]
+        self.selected_module = None
+        self.graph_json = None
+        self.lib = None
+        self.params = {}
+        self.iter_cnt = 0
+        super(GraphRuntimeFactoryModule, self).__init__(self.module.handle)
+
+    def __del__(self):
+        pass
+
+    def runtime_create(self, ctx):
+        """Create the runtime using ctx
+
+        Parameters
+        ----------
+        ctx : TVMContext or list of TVMContext
+        """
+        ctx, num_rpc_ctx, device_type_id = get_device_ctx(self.selected_module, ctx)
+        if num_rpc_ctx == len(ctx):
+            fcreate = ctx[0]._rpc_sess.get_function("tvm.graph_runtime_factory.runtime_create")
+        else:
+            fcreate = get_global_func("tvm.graph_runtime_factory.runtime_create")
+        return fcreate(self.selected_module, *device_type_id)
+
+    def import_module(self, mod):
+        """Create the runtime using ctx
+
+        Parameters
+        ----------
+        mod : GraphRuntimeFactoryModule
+            The graph runtime factory module we want to import
+        """
+        return self._import_module(mod)
+
+    def __getitem__(self, key='default'):
+        """Get specific module
+
+        Parameters
+        ----------
+        key : str
+            The key of module.
+        """
+        self.selected_module = self._select_module(key)
+        self.selected_module._entry = self.runtime_create
+        return self.selected_module
+
+    def __iter__(self):
+        warnings.warn(
+            "legacy graph runtime behaviour of producing json / lib / params will be removed in the next release ",
+            DeprecationWarning, 2)
+        self.graph_json = self.module["get_json"]()
+        self.lib = self.module["get_lib"]()
+        for k, v in self.module["get_params"]().items():
+            self.params[k] = v
+        return self
+
+
+    def __next__(self):
```

Comment:
I am sorry. I am little lost here. If it is possible , would you please share one example, how you are planning to use this iter & next feature?

Reply:
Please see the example of [test_legacy_compatibility](https://github.com/apache/incubator-tvm/blob/784a6bdbecc0ff3040d003c90ede5794b8be816a/tests/python/unittest/test_module_runtime_interface.py#L42). In the new design, we like to only export `compiled graph module`. However, we also want to maintain legacy compatibility in one release of TVM (and remove it in the next release). So, `__iter__` / `__next__` help us do this thing.